src/main/java/Duke/Duke.java

```diff
@@ -0,0 +1,144 @@
+package Duke;
+
+import Duke.Functions.*;
+import Duke.Exceptions.DukeException;
+import Duke.Task.*;
+
+import java.util.List;
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.io.*;
+
+public class Duke {
+    private static String LINE = "____________________________________________________________";
+    public static String inputString;
+    public static int taskCount = 0;
+    public static List<Task> lists = new ArrayList<Task>();
+
+    public static void main(String[] args) {
+        greetings();
+        readFile(lists);
+        processInput();
+        writeFile(lists);
+        goodbye();
+    }
+
+
+    public static void processInput(){
+        Scanner userInput = new Scanner(System.in);
+        while(true) {
+            inputString = userInput.nextLine().trim();
+            String[] inputStringSplit = inputString.split(" ", 2);
+            System.out.println(LINE);
+            if (inputString.equalsIgnoreCase("bye")) {
+                return;
+            } else if (inputStringSplit[0].equalsIgnoreCase("done")) {
+                if (inputStringSplit.length > 1) {
+                    MarkAsDone.markAsDone(inputStringSplit[1]);
+                } else {
+                    DukeException.doneWithoutNo();
+                }
+            } else if (inputStringSplit[0].equalsIgnoreCase("delete")) {
+                if (inputStringSplit.length > 1) {
+                    Delete.deleteTask(inputStringSplit[1]);
+                } else {
+                    DukeException.deleteWithoutNo();
+                }
+            } else if (inputStringSplit[0].equalsIgnoreCase("list")) {
+                printList(0, taskCount);
+            } else if (AddToList.keywordCheck(inputStringSplit[0])) {
+                if (inputStringSplit.length > 1) {
+                    AddToList.AddToList(inputStringSplit[0], inputStringSplit[1]);
+                } else {
+                    DukeException.taskDescriptionEmpty();
+                }
+            } else {
+                DukeException.illegalInput();
+            }
+        }
+    }
+
+    private static void readFile(List<Task> lists){
```

Comment:
Perhaps you could restructure your functions to avoid deep if-else nesting, to increase its readability and make debugging easier for yourself!